### PR TITLE
Fix E2Es for RHS and Accessibility  (Advanced Text Editor)

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
Fix E2Es for RHS and Accessibility (Advanced Text Editor)

#### Ticket Link
 Fixes https://mattermost.atlassian.net/browse/MM-44990
~Fixes https://mattermost.atlassian.net/browse/MM-44977~ removed `/poll` ticket from this PR
 Fixes https://mattermost.atlassian.net/browse/MM-44970
 Fixes https://mattermost.atlassian.net/browse/MM-44968
 Fixes https://mattermost.atlassian.net/browse/MM-44985
 Fixes https://mattermost.atlassian.net/browse/MM-44991
 
#### Related Pull Requests
NONE 

#### Screenshots
NONE

#### Release Note
```release-note
Fix: E2Es for RHS and Accessibility (Advanced Text Editor)
```

### CODE REVIEW

**Comment by AshishDhama** [comment]
@jgilliam17 Yeah there was some issue have fixed that now.

**Review by michelengelen — CHANGES_REQUESTED** [review]
looking good overall .. just some suggestions.

What really needs to be adjusted are the descriptions for the test-steps.

File: e2e/cypress/tests/support/ui/sidebar_right.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ Cypress.Commands.add('isExpanded', {prevSubject: true}, (subject) => {
 });
 
 Cypress.Commands.add('uiGetReply', () => {
-    return cy.findByRole('button', {name: 'Reply'});
+    return cy.get('.sidebar--right').findByTestId('SendMessageButton');
```
[the remark this answers is not in the document]
Comment (AshishDhama) [reply e2e/cypress/tests/support/ui/sidebar_right.js:25]: Yeah will add this in future PR

File: e2e/cypress/tests/integration/enterprise/accessibility/accessibility_input_fields_spec.js [hunk 2]
```diff
@@ -159,41 +159,41 @@ describe('Verify Accessibility Support in different input fields', () => {
     });
 
     it('MM-T1458 Verify Accessibility Support in Main Post Input', () => {
-        cy.get('#centerChannelFooter').within(() => {
+        cy.get('#advancedTextEditorCell').within(() => {
```
[the remark this answers is not in the document]
Comment (michelengelen) [reply e2e/cypress/tests/integration/enterprise/accessibility/accessibility_input_fields_spec.js:162]: yep, after going through the changes: we use that a lot ... so could you please create a getter function for that?

**Review by michelengelen — CHANGES_REQUESTED** [review]
still a few things that need to change, but it is looking great for the most part!

**Review by furqanmlk — COMMENTED** [review]
Thank @AshishDhama  Following test cases files are failing in your spec files.
image

image

File: e2e/cypress/tests/integration/enterprise/accessibility/accessibility_input_fields_spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (AshishDhama) [reply e2e/cypress/tests/integration/enterprise/accessibility/accessibility_input_fields_spec.js:162]: Yeah will add this in next separate PR

**Review by jgilliam17 — APPROVED** [review]
Thanks @AshishDhama 
Here is the latest [E2E report](https://mattermost-cypress-report.s3.amazonaws.com/210283-24ae0ed-onprem-ent-webapp-pr-10586-mattermost/mm-ee-test24ae0ed/mochawesome.html), all listed test cases are passing, except `/poll` so I will remove it from the description and we can address in a different PR.

**Review by furqanmlk — APPROVED** [review]
Thanks @AshishDhama LGTM

**Comment by AshishDhama** [comment]
@michelengelen Please review it as soon as possible if there aren’t any blocking changes would like to merge it, I need changes done on this pr to fix other breaking e2e don’t want to make this a giant with 100 file changes. Please add your suggestions if any will add those in follow up PR.

**Review by michelengelen — COMMENTED** [review]
@AshishDhama there are still some things that haven't been adressed yet. Other than that this looks good to merge for me

**Comment by michelengelen** [comment]
> @AshishDhama there are still some things that haven't been adressed yet. Other than that this looks good to merge for me

I don't know why, but when I was looking now the files had the changes I was expecting... For some reason it did not update it when I had a look some minutes ago. So everything is fine... I'll approve and merge!

Sorry for the false alarm! 🙇🏼

**Review by michelengelen — APPROVED** [review]

**Pull request merged**